File: gdscript/src/main/java/gdscript/psi/GdClassDeclTl.java
package gdscript.psi;

import com.intellij.psi.StubBasedPsiElement;
import gdscript.index.stub.GdClassDeclStub;
import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface GdClassDeclTl extends GdTopLevelDecl, StubBasedPsiElement<GdClassDeclStub>, GdDocumented {

  @Nullable
  GdClassNameNmi getClassNameNmi();

  @NotNull
  List<GdInheritance> getInheritanceList();

  @NotNull
  List<GdTopLevelDecl> getTopLevelDeclList();

  @NotNull
  String getName();

  @NotNull
  String getParentName();

}

File: gdscript/src/main/kotlin/gdscript/psi/GdClassDeclTl.kt
package gdscript.psi

import com.intellij.psi.StubBasedPsiElement
import gdscript.index.stub.GdClassDeclStub
import gdscript.psi.types.GdDocumented

interface GdClassDeclTl : GdTopLevelDecl, StubBasedPsiElement<GdClassDeclStub>, GdDocumented {
    val classNameNmi: GdClassNameNmi?

    val inheritanceList: List<GdInheritance>

    val topLevelDeclList: List<GdTopLevelDecl>

    fun getName(): String

    val parentName: String
}
